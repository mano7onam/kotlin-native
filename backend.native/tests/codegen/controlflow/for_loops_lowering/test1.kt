fun foo() : Int {
    return 2
}

fun main(args: Array<String>) {
    val rng = 5 .. 10
    var arr = arrayOf(1, 2, 3)
    for (i in rng) {
    	for (j in arr.indices) {
          for (k in 4 downTo 0 step foo()) {
              println("$i + $j = ${i + j}")
          }
      }
    }
}