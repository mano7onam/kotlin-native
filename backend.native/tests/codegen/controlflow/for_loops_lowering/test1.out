5 + 0 = 5
5 + 0 = 5
5 + 0 = 5
5 + 1 = 6
5 + 1 = 6
5 + 1 = 6
5 + 2 = 7
5 + 2 = 7
5 + 2 = 7
6 + 0 = 6
6 + 0 = 6
6 + 0 = 6
6 + 1 = 7
6 + 1 = 7
6 + 1 = 7
6 + 2 = 8
6 + 2 = 8
6 + 2 = 8
7 + 0 = 7
7 + 0 = 7
7 + 0 = 7
7 + 1 = 8
7 + 1 = 8
7 + 1 = 8
7 + 2 = 9
7 + 2 = 9
7 + 2 = 9
8 + 0 = 8
8 + 0 = 8
8 + 0 = 8
8 + 1 = 9
8 + 1 = 9
8 + 1 = 9
8 + 2 = 10
8 + 2 = 10
8 + 2 = 10
9 + 0 = 9
9 + 0 = 9
9 + 0 = 9
9 + 1 = 10
9 + 1 = 10
9 + 1 = 10
9 + 2 = 11
9 + 2 = 11
9 + 2 = 11
10 + 0 = 10
10 + 0 = 10
10 + 0 = 10
10 + 1 = 11
10 + 1 = 11
10 + 1 = 11
10 + 2 = 12
10 + 2 = 12
10 + 2 = 12
